python/lsst/summit/utils/astrometry/utils.py

```diff
@@ -0,0 +1,288 @@
+
+    plateScale = header.get('SECPIX')
+    if not plateScale:
+        raise RuntimeError('Failed to find platescale in header')
+    plateScale = float(plateScale)
+
+    header['CD1_1'] = plateScale / 3600
+    header['CD1_2'] = 0
+    header['CD2_1'] = 0
+    header['CD2_2'] = plateScale / 3600
+
+    wcsPropSet = PropertySet.from_mapping(header)
+    wcs = SkyWcs(wcsPropSet)
+    return wcs
+
+
+def getIcrsAtZenith(lon, lat, height, utc):
+    """Get the icrs at zenith given a lat/long/height/time in UTC.
+
+    Parameters
+    ----------
+    lon : `float`
+        The longitude, in degrees.
+    lat : `float`
+        The latitude, in degrees.
+    height : `float`
+        The height above sea level in meters.
+    utc : `str`
+        The time in UTC as an ISO string, e.g. '2022-05-27 20:41:02'
+
+    Returns
+    -------
+    skyCoordAtZenith : `astropy.coordinates.SkyCoord`
+        The skyCoord at zenith.
+    """
+    location = EarthLocation.from_geodetic(lon=lon*u.deg,
+                                           lat=lat*u.deg,
+                                           height=height)
+    obsTime = Time(utc, format='iso', scale='utc')
+    skyCoord = SkyCoord(AltAz(obstime=obsTime,
+                              alt=90.0*u.deg,
+                              az=180.0*u.deg,
+                              location=location))
+    return skyCoord.transform_to('icrs')
+
+
+def headerToWcs(header):
+    """Convert an astrometry.net wcs header dict to a DM wcs object.
+
+    Parameters
+    ----------
+    header : `dict`
+        The wcs header, as returned from from the astrometry_net fit.
+
+    Returns
+    -------
+    wcs : `lsst.afw.geom.SkyWcs`
+        The wcs.
+    """
+    wcsPropSet = PropertySet.from_mapping(header)
+    return SkyWcs(wcsPropSet)
+
+
+def runCharactierizeImage(exp, snr, minPix):
+    """Run the image characterization task, finding only bright sources.
+
+    Parameters
+    ----------
+    exp : `lsst.afw.image.Exposure`
+        The exposure to characterize.
+    snr : `float`
+        The SNR threshold for detection.
+    minPix : `int`
+        The minimum number of pixels to count as a source.
+
+    Returns
+    -------
+    result : `lsst.pipe.base.Struct`
+        The result from the image characterization task.
+    """
+    charConfig = CharacterizeImageConfig()
+    charConfig.doMeasurePsf = False
+    charConfig.doApCorr = False
+    charConfig.doDeblend = False
+    charConfig.repair.doCosmicRay = False
+    charConfig.repair.doInterpolate = True
+    charConfig.detection.minPixels = minPix
+    charConfig.detection.thresholdValue = snr
+
+    charTask = CharacterizeImageTask(config=charConfig)
+
+    charResult = charTask.run(exp)
+    return charResult
+
+
+def filterSourceCatOnBrightest(catalog, brightFraction, minSources=15,
+                               flux_field="base_CircularApertureFlux_3_0_instFlux"):
+    """Return a catalog containing the brightest sources in the input,
```

Comment:
Short description should be one line with more details below.

Reply:
Thanks.